### PR TITLE
Update TagDispatcher.java

### PR DESCRIPTION
**Pull request by adriaan-telcred**
If adapter is disabled, prompt user to enable NFC (show NFC settings).

### CODE REVIEW

**Comment by Franzaine** [comment]
From what I can see this doesn't prompt the user to enable NFC, it just sends them to the setting. There should at least be a Toast telling the user that they need to activate NFC.

Sidenote: I think this is a good enough solution for now but other than that I'd like to see an endpoint/callback instead that allows the implementation to handle this themselves. Maybe it is not always wanted to simply send the user to the settings but rather deliver using some other method or simply handle this in some other way.

**Comment by adriaan-telcred** [comment]
@Franzaine @petter-fidesmo  

I like the idea of a callback - perhaps extend OnDiscoveredTagListener with a method onError() ?

One could add a static method in TagDispatcher for bringing up the NFC settings activity; this could be called from the error callback.

In any case I didn't want to put in a toast since that should contain some localised strings, which haven't been added to the project.

**Comment by Franzaine** [comment]
Sounds good to me from a UX perspective :) But I have no idea about the actual implementation. @slomo or @petter-fidesmo do what do you think?

File: android/src/main/java/nordpol/TagDispatcher.java [hunk 1]
```diff
@@ -62,6 +69,14 @@ public boolean disableExclusiveNfc() {
         return false;
     }
 
+        /** Call the TagDispatcher's listener.
```
Comment (slomo) [inline android/src/main/java/nordpol/TagDispatcher.java]: I think this should only be indented 4 spaces

Comment (slomo) [inline android/src/main/java/nordpol/TagDispatcher.java]: Can you indent this section only 4 spaces

**Comment by slomo** [comment]
The rest seems quite nice. Could you please rebase until the current master, it should also fix the travis build.

Sorry for the nitpicking about indenting and so on, but we try to enforce [googles java coding style](http://google.github.io/styleguide/javaguide.html).

**Comment by slomo** [comment]
I also like the callback idea. But lets try to make small steps here and merge this first. Than we  can have a look how well it integrates into our apps.

File: android/src/main/java/nordpol/TagDispatcher.java (hunk 1 above)
Comment (adriaan-telcred) [inline android/src/main/java/nordpol/TagDispatcher.java]: @slomo Done.

**Comment by adriaan-telcred** [comment]
@slomo is the short summary of commit messages the single message in the conversation root, or the nested ones under that? Terminology unclear ...

**Comment by slomo** [comment]
I intended to talk about the title

**Comment by adriaan-telcred** [comment]
@slomo OK, will keep it in mind!

File: android/src/main/java/nordpol/TagDispatcher.java [hunk 2]
```diff
@@ -12,6 +12,8 @@
 import android.os.Bundle;
 
 public class TagDispatcher {
+    private static final int DELAY_PRESENCE = 60000;
```
Comment (adriaan-telcred) [inline android/src/main/java/nordpol/TagDispatcher.java]: @slomo @petter-fidesmo  Is a presence check even applicable to NFC A tags used in IsoDep mode? 

I'm in favour of disabling the presence check completely...

**Comment by petter-fidesmo** [comment]
I like the idea of the callback, I think the following solution would support most use cases and provide a good default behaviour:

Let's add an interface OnNfcDisabledListener:
```JAVA
public interface OnNfcDisabledListener {
    public void nfcDisabled();
}
```
Then let's pass this as an additional parameter to the constructor (and the factory method). If provided, we let `enableExclusiveNfc` return false if NFC is disabled and call the listener, if it is not provided (i.e. null), as a default behaviour we pop up the settings dialogue. Of course, we also provide a simplified factory method without the listener parameter that just defaults it to null. We can also add a toast as soon as I have set up the localized strings for the project.

I would be in favour of merging this if the solution provided this flexibility.

@adriaan-telcred What do you think?

**Comment by adriaan-telcred** [comment]
@petter-fidesmo Sounds good!

File: android/src/main/java/nordpol/TagDispatcher.java [hunk 3]
```diff
@@ -12,6 +12,8 @@
 import android.os.Bundle;
 
 public class TagDispatcher {
+    private static final int DELAY_PRESENCE = 5000;
```
Comment (adriaan-telcred) [inline android/src/main/java/nordpol/TagDispatcher.java]: @petter-fidesmo I can submit the "keepalive" workaround if you are interested.

File: android/src/main/java/nordpol/TagDispatcher.java (hunk 2 above)
Comment (petter-fidesmo) [inline android/src/main/java/nordpol/TagDispatcher.java]: @adriaan-telcred Yes, sadly it is. Please see: https://code.google.com/p/android/issues/detail?id=58773

File: android/src/main/java/nordpol/TagDispatcher.java (hunk 3 above)
Comment (petter-fidesmo) [inline android/src/main/java/nordpol/TagDispatcher.java]: @adriaan-telcred Please see: https://code.google.com/p/android/issues/detail?id=58773

We would need to identify if the phone is using a broadcom chipset to be able to deactivate this timeout.

Comment (adriaan-telcred) [inline android/src/main/java/nordpol/TagDispatcher.java]: @petter-fidesmo That would require a very extensive phone model database and name matching rules - not worth the effort at this point in time :)

**Comment by petter-fidesmo** [comment]
@adriaan-telcred One last thing. Could you also rebase the branch and squash some committs. Since the change set is fairly limited 1 or 2 commits should be enough. This is a great guide on rebasing: https://github.com/edx/edx-platform/wiki/How-to-Rebase-a-Pull-Request

**Comment by adriaan-telcred** [comment]
@petter-fidesmo Squashed and rebased!

**Comment by petter-fidesmo** [comment]
@adriaan-telcred I tried to compile it locally and there is a syntax error. I realized that we are missing a description on how to build the project locally. https://github.com/fidesmo/nordpol/blob/feature-add-description-of-how-to-build/README.md in this pull request #20 

Maybe you can try those instructions and see if you can compile it locally and then fix the syntax error?

**Comment by adriaan-telcred** [comment]
Compilation errors have been fixed, and I rebased my patch to fidesmo:master 66a17f150d8087565dde7408ee2e507e41eff663.

**Comment by petter-fidesmo** [comment]
@adriaan-telcred Great, I'd say we are really close now, but just two last things. Would it be possible to squash the last commit into the previous one? Also the comment @slomo left regarding the comment being oddly indented also needs to be addressed. Comments should be indented at the same level as the method, in this case 4 space characters.

**Comment by adriaan-telcred** [comment]
@petter-fidesmo @slomo Indentation fixed, and squashed again.

**Comment by petter-fidesmo** [comment]
@adriaan-telcred :+1: This is looking great! Thank you for your efforts and sorry for the trouble with the build. I will update the documentation with the information needed.

**Comment by adriaan-telcred** [comment]
:+1:

**Pull request merged**